### DISCUSSION TITLE
Migration to jakarta

### DISCUSSION THREAD

**Discussion by Siggen** [jorabin/KeePassJava2#58] [Ideas]
Hello @jorabin ,

Do you have any plan to migrate KeePassJava2 to Jakarta namespace ("jakarta.xml.bind" instead of "javax.xml.bind") ? If not, I may give a try...

Thanks in advance.

**Comment by jorabin** [comment]
Hi @Siggen 

Short answer "no", longer answer is that i'm not clear of the benefit of keeping the JAXB implementation going. The Simple version won't work in Java 17 and beyond and the DomWrapper version is very slow. Version 2.2.2 will feature a Jackson Implementation and I think we should all think about moving to that - I'm also working (very much as a background task) on a version 3.

Keen to understand why you favour the JAXB implementation?

thanks

**Reply by Siggen** [reply]
Thanks for you response, @jorabin.

The jackson implementation looks promising and it should work nicely with jakarta namespaces. I'll have a look at it and I'll let you know.

I have no special favour for the JAXB implementation, it's just I missed the jackson one :-)

For what it may worth, I had a quick shot at migrating to java17 and jakarta [here](https://github.com/Siggen/KeePassJava2/tree/jakarta). Its pretty much a work-in-progress (but it compiles fine) but I don't think I'll spend more time on it unless you think it may somehow be useful.

Best regards.

**Comment by Siggen** [comment]
The Jackson Implementation (version 2.2.2-SNAPSHOT) works nicely with jakarta namespace, thank you !

Do you plan to make a stable version in the near future ?

**Comment by jorabin** [comment]
I planned to do it a couple of months ago, but it can't have been a good plan, can it! Current plan is to do it late December when hopefully it will be a little quieter than it has been.

Best

**Reply by Siggen** [reply]
Thanks ! Looking forward to it :-)

**Comment by jorabin** [comment]
Noting that 2.2.2 has been released (finally)

**Comment by jorabin** [comment]
Given release of 2.2.2 with a Jackson implementation and withdrawal of JAXB in planned version 3 #62 there is no longer any dependency on javax so this is moot I think.
